draco/tests/test_hard.py

```diff
@@ -319,3 +320,596 @@ def test_point_tick_bar_without_x_or_y():
         )
         == ["point_tick_bar_without_x_or_y"]
     )
+
+
+def test_enc_type_valid():
+    b = hard.blocks["enc_type_valid"]
+    assert isinstance(b, Block)
+
+    assert no_violations(
+        b.program
+        + """
+    attribute((field,type),temperature,number).
+    attribute((field,type),name,string).
+    entity(mark,root,1).
+    entity(encoding,1,3).
+    entity(scale,root,5).
+    attribute((encoding,channel),3,x).
+    attribute((encoding,field),3,temperature).
+    attribute((scale,channel),5,x).
+    attribute((scale,type),5,linear).
+    """
+    )
+
+    assert (
+        list_violations(
+            b.program
+            + """
+    attribute((field,type),temperature,string).
+    entity(mark,0,1).
+    entity(encoding,1,3).
+    entity(scale,0,5).
+    attribute((encoding,channel),3,x).
+    attribute((encoding,field),3,temperature).
+    attribute((scale,channel),5,x).
+    attribute((scale,type),5,linear).
+    """
+        )
+        == ["enc_type_valid"]
+    )
+
+    # shared scale
```

Comment:
Unclear why it's invalid if we share a scale.

Reply:
same with the previous case. only the structure is different(with/without a shared scale)